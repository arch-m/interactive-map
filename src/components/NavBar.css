#sidebar {
    min-width: 250px;
    max-width: 250px;
    max-height: 100vh;
    z-index: 1;
}

a, a:hover, a:focus{
    text-decoration: none;
}

#sidebar a {
    display: block;
    width: 100%;
    background-color: #d0c1fa;
    padding: 0.5rem 2rem;
    margin-bottom: 0.4rem;
    font-size: 1.2rem ;
    border-radius: 0.2rem;
    color: #3d2e8b;
    animation: slideBack ease 0.5s;
    -webkit-animation: slideBack ease 0.5s;
    -moz-animation: slideBack ease 0.5s;
    -o-animation: slideBack ease 0.5s;
    -ms-animation: slideBack ease 0.5s;
}

.sidebar-header{
    color: #c6f8fc;
}

.sidebar-footer{
    color: #6C1D45 !important;
}

.toggler:focus{
    border: none !important;
}

#sidebar-icon{
    height: calc(1.3rem + 0.6vw);
}

@keyframes slideBack {
    0% {margin-left: 20px;}
    100% {margin-left: 0;}
}

@-moz-keyframes slideBack {
    0% {margin-left: 20px;}
    100% {margin-left: 0;}
}

@-webkit-keyframes slideBack {
    0% {margin-left: 20px;}
    100% {margin-left: 0;}
}

@-o-keyframes slideBack {
    0% {margin-left: 20px;}
    100% {margin-left: 0;}
}

@-ms-keyframes slideBack {
    0% {margin-left: 20px;}
    100% {margin-left: 0;}
}

#sidebar a:hover {
    margin-left: 20px;
    color: #ece8ff;
    background-color: #6751a3;
    animation: slideRight ease 0.1s;
    -webkit-animation: slideRight ease 0.1s;
    -moz-animation: slideRight ease 0.1s;
    -o-animation: slideRight ease 0.1s;
    -ms-animation: slideRight ease 0.1s;
}

#sidebar a:focus{
    background-color: #8551a3;
    color: #ece8ff;
}

@keyframes slideRight {
    0% {margin-left: 0;}
    100% {margin-left: 20px;}
}

@-moz-keyframes slideRight {
    0% {margin-left: 0;}
    100% {margin-left: 20px;}
}

@-webkit-keyframes slideRight {
    0% {margin-left: 0;}
    100% {margin-left: 20px;}
}

@-o-keyframes slideRight {
    0% {margin-left: 0;}
    100% {margin-left: 20px;}
}

@-ms-keyframes slideRight {
    0% {margin-left: 0;}
    100% {margin-left: 20px;}
}






