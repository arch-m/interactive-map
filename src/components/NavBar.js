import React from 'react';
import {
  Link
} from "react-router-dom";

export const NavBar = () => {
  return (
    <>
      <div>
        <nav>
          <ul>
            <li>
              <Link to="/">Mapa</Link>
            </li>
            <li>
              <Link to="/cuestionarios">Cuestionarios</Link>
            </li>            
            <li>
              <Link to="/referencias">Referencias</Link>
            </li>
            {/* <li>
              <Link to="/enciclopedia">Enciclopedia</Link> Componente pendiente por dificultad de realizacion, necesita filtros y
              queries para la busqueda de datos, ademas de necesitar uso de hooks de memorizacion para efeciencia
            </li> */}
          </ul>
        </nav>
      </div>
    </>
  );
}