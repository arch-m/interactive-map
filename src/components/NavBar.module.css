.btn{
    color: #c6f8fc;
    background-color: transparent;
    fill: #c6f8fc;
    font-size: 2.3em;
    padding: 0;
    margin: 0;
    line-height: 0%;
    transition: 0.2s ease;
}

.btn path:hover{
    fill: #4c98bb !important;
}

.btn:hover{
    color: #4c98bb;
    background-color: #c6f8fc;
    transition: 0.2s ease;
}

.btn:focus{
    outline: none;
}

.active{
    margin-left: -197px !important;
    transition: ease  0.5s ;
}

.show{
    transition: ease 0.5s;
}

.hidden{
    animation: fadeOut ease 0.2s;
    -webkit-animation: fadeOut ease 0.2s;
    -moz-animation: fadeOut ease 0.2s;
    -o-animation: fadeOut ease 0.2s;
    -ms-animation: fadeOut ease 0.2s;
    opacity: 0;
}

@keyframes fadeOut {
    0% {opacity: 1;}
    100% {opacity: 0;}
}

@-moz-keyframes fadeOut {
    0% {opacity: 1;}
    100% {opacity: 0;}
}

@-webkit-keyframes fadeOut {
    0% {opacity: 1;}
    100% {opacity: 0;}
}

@-o-keyframes fadeOut {
    0% {opacity: 1;}
    100% {opacity: 0;}
}

@-ms-keyframes fadeOut {
    0% {opacity: 1;}
    100% {opacity: 0;}
}