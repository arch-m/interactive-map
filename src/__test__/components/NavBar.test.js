import { shallow } from "enzyme";
import { NavBar } from "../../components/NavBar";

describe('Pruebas a NavBar', () => {

  const wrapper = shallow(<NavBar />);

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h3').text().trim()).toBe('AWA');
  });
  
});
