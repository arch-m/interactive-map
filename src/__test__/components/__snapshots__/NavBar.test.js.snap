// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a NavBar Debe mostrarse correctamente 1`] = `
<Fragment>
  <nav
    className="show d-flex flex-column bg-light"
    id="sidebar"
  >
    <div
      className="sidebar-header p-2 d-flex bg-secondary"
    >
      <div
        className="ms-1 d-flex justify-content-center align-items-center"
      >
        <img
          alt="Icono de AWA"
          id="sidebar-icon"
          src="awa2.png"
        />
        <h3
          className="m-0 mb-1 px-2"
        >
          AWA
        </h3>
      </div>
      <div
        className="ms-auto"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          <RiMenuFill />
        </button>
      </div>
    </div>
    <div
      className="p-2 false"
    >
      <div
        className="text-secondary p-2 my-3"
      >
        <h4>
          Geografía de México
        </h4>
      </div>
      <ul
        className="list-unstyled components"
      >
        <li>
          <Link
            to="/"
          >
            Mapa Interactivo
          </Link>
        </li>
        <li>
          <Link
            to="/cuestionarios"
          >
            Cuestionarios
          </Link>
        </li>
        <li>
          <Link
            to="/referencias"
          >
            Referencias
          </Link>
        </li>
      </ul>
    </div>
    <div
      className="sidebar-footer mt-auto px-2 text-secondary false"
    >
      <h5>
        Instituto Politécnico Nacional 
      </h5>
      <p
        className="mb-1"
      >
        Escuela Superior de Cómputo
      </p>
      <p>
        2021
      </p>
    </div>
  </nav>
</Fragment>
`;
